File: QE-Selenium-Java/Activity7.java
package demos;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity7 {

    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));

        try { 
            driver.get("https://training-support.net/webelements/dynamic-controls");

            System.out.println("Page Title: " + driver.getTitle());

            WebElement inputField = driver.findElement(By.cssSelector("input[type='text']"));

            System.out.println("Input enabled before click: " + inputField.isEnabled());

            WebElement enableButton = driver.findElement(By.xpath("//button[text()='Enable Input']"));
            enableButton.click();

            wait.until(ExpectedConditions.elementToBeClickable(inputField));

            System.out.println("Input enabled after click: " + inputField.isEnabled());

        } finally {
            driver.quit();
        }
    }
}

File: QE-Selenium-Python/Activity7.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

driver = webdriver.Firefox()
wait = WebDriverWait(driver, 10)

try:
    driver.get("https://training-support.net/webelements/dynamic-controls")

    print("Page Title:", driver.title)

    input_field = driver.find_element(By.CSS_SELECTOR, "input[type='text']")

    print("Input enabled before click:", input_field.is_enabled())

    enable_button = driver.find_element(By.XPATH, "//button[contains(text(),'Enable')]")
    enable_button.click()

    wait.until(EC.element_to_be_clickable((By.CSS_SELECTOR, "input[type='text']")))

    print("Input enabled after click:", input_field.is_enabled())

finally:
    driver.quit()
